File: packages/features/feature_library/lib/src/detail/word_detail_screen.dart
import 'package:core_designsystem/core_designsystem.dart';
import 'package:core_domain/core_domain.dart';
import 'package:core_ui/core_ui.dart';
import 'package:flutter/material.dart';
import 'package:flutter_hooks/flutter_hooks.dart';
import 'package:hooks_riverpod/hooks_riverpod.dart';
import 'word_detail_notifier.dart';

class WordDetailScreen extends HookConsumerWidget {
  const WordDetailScreen({super.key, required this.wordId});

  final String wordId;

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final state = ref.watch(wordDetailProvider(wordId));
    final notifier = ref.read(wordDetailProvider(wordId).notifier);

    if (state.isLoading || state.currentWord == null) {
      return const Scaffold(
        body: Center(child: CircularProgressIndicator()),
      );
    }

    final contextIds = state.contextIds;
    final initialIndex = contextIds.indexOf(wordId);
    final pageController = usePageController(initialPage: initialIndex >= 0 ? initialIndex : 0);

    return Scaffold(
      backgroundColor: Theme.of(context).colorScheme.background,
      body: PageView.builder(
        controller: pageController,
        itemCount: contextIds.length,
        itemBuilder: (context, index) {
          final currentId = contextIds[index];
          // We can use another provider or just fetch from notifier if it's sync mock logic
          final word = (currentId == wordId) ? state.currentWord : notifier.getWordById(currentId);
          
          if (word == null) return const Center(child: CircularProgressIndicator());
          
          return _WordDetailContent(
            word: word,
            playingAudioId: state.playingAudioId,
            onPlayAudio: notifier.playAudio,
            onBack: () => Navigator.of(context).pop(),
          );
        },
      ),
    );
  }
}

class _WordDetailContent extends StatelessWidget {
  const _WordDetailContent({
    required this.word,
    required this.playingAudioId,
    required this.onPlayAudio,
    required this.onBack,
  });

  final Word word;
  final String? playingAudioId;
  final Function(String, String) onPlayAudio;
  final VoidCallback onBack;

  @override
  Widget build(BuildContext context) {
    final isDark = Theme.of(context).brightness == Brightness.dark;
    final primaryColor = Theme.of(context).colorScheme.primary;

    return SingleChildScrollView(
      child: Column(
        children: [
          // === 1. Immersive Hero Section ===
          Stack(
            children: [
              Container(
                width: double.infinity,
                padding: EdgeInsets.only(
                  top: MediaQuery.of(context).padding.top + 56,
                  bottom: 32,
                  left: 24,
                  right: 24,
                ),
                decoration: BoxDecoration(
                  gradient: LinearGradient(
                    begin: Alignment.topCenter,
                    end: Alignment.bottomCenter,
                    colors: [
                      primaryColor.withOpacity(isDark ? 0.2 : 0.1),
                      Theme.of(context).colorScheme.background,
                    ],
                  ),
                ),
                child: Column(
                  children: [
                    // Kanji
                    Text(
                      word.japanese,
                      style: Theme.of(context).textTheme.displayMedium?.copyWith(
                            fontWeight: FontWeight.bold,
                            letterSpacing: 1.0,
                            color: Theme.of(context).colorScheme.onBackground,
                          ),
                    ),
                    const SizedBox(height: 8),
                    // Hiragana
                    Text(
                      word.hiragana,
                      style: Theme.of(context).textTheme.titleLarge?.copyWith(
                            fontWeight: FontWeight.w500,
                            color: Theme.of(context).colorScheme.onBackground.withOpacity(0.7),
                          ),
                    ),
                    const SizedBox(height: 24),
                    // Speaker Button
                    NemoSpeakerButton(
                      isPlaying: playingAudioId == 'word_${word.id}',
                      size: 56,
                      backgroundColor: Theme.of(context).colorScheme.primaryContainer,
                      tint: Theme.of(context).colorScheme.onPrimaryContainer,
                      onClick: () => onPlayAudio(word.japanese, 'word_${word.id}'),
                    ),
                  ],
                ),
              ),
              // Back Button Overlay
              Positioned(
                top: MediaQuery.of(context).padding.top,
                left: 8,
                child: IconButton(
                  icon: const Icon(Icons.arrow_back_ios_new_rounded),
                  onPressed: onBack,
                ),
              ),
            ],
          ),

          // === 2. Meaning & Tags ===
          Padding(
            padding: const EdgeInsets.symmetric(horizontal: 20),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.stretch,
              children: [
                PremiumCard(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(
                        '中文释义',
                        style: Theme.of(context).textTheme.labelLarge?.copyWith(
                              color: primaryColor,
                              fontWeight: FontWeight.bold,
                            ),
                      ),
                      const SizedBox(height: 8),
                      Text(
                        word.chinese,
                        style: Theme.of(context).textTheme.headlineSmall?.copyWith(
                              fontWeight: FontWeight.bold,
                            ),
                      ),
                      const SizedBox(height: 20),
                      // Tags Row
                      Row(
                        children: [
                          _DetailTag(
                            text: word.level,
                            backgroundColor: NemoColors.brandBlue,
                            textColor: Colors.white,
                          ),
                          const SizedBox(width: 8),
                          if (word.pos != null)
                            _DetailTag(
                              text: word.pos!,
                              backgroundColor: Theme.of(context).colorScheme.secondaryContainer,
                              textColor: Theme.of(context).colorScheme.onSecondaryContainer,
                            ),
                        ],
                      ),
                    ],
                  ),
                ),

                const SizedBox(height: 24),

                // === 3. Example Sentences ===
                Padding(
                  padding: const EdgeInsets.only(left: 4, bottom: 12),
                  child: Text(
                    '例句',
                    style: Theme.of(context).textTheme.titleLarge?.copyWith(
                          fontWeight: FontWeight.bold,
                        ),
                  ),
                ),

                if (word.examples.isEmpty)
                  Padding(
                    padding: const EdgeInsets.only(left: 4),
                    child: Text(
                      '暂无例句',
                      style: Theme.of(context).textTheme.bodyMedium?.copyWith(
                            color: Theme.of(context).colorScheme.onSurfaceVariant,
                          ),
                    ),
                  )
                else
                  ...word.examples.asMap().entries.map((entry) {
                    final index = entry.key + 1;
                    final example = entry.value;
                    return Padding(
                      padding: const EdgeInsets.only(bottom: 12),
                      child: _ExampleCard(
                        index: index,
                        example: example,
                        isPlaying: playingAudioId == 'example_${word.id}_$index',
                        onPlay: () => onPlayAudio(example.japanese, 'example_${word.id}_$index'),
                      ),
                    );
                  }),

                const SizedBox(height: 48),
              ],
            ),
          ),
        ],
      ),
    );
  }
}

// Removed _HeroSpeakerButton since we use NemoSpeakerButton

class _DetailTag extends StatelessWidget {
  const _DetailTag({
    required this.text,
    required this.backgroundColor,
    required this.textColor,
  });

  final String text;
  final Color backgroundColor;
  final Color textColor;

  @override
  Widget build(BuildContext context) {
    return Container(
      height: 28,
      padding: const EdgeInsets.symmetric(horizontal: 12),
      decoration: BoxDecoration(
        color: backgroundColor,
        borderRadius: BorderRadius.circular(14),
      ),
      child: Center(
        child: Text(
          text,
          style: Theme.of(context).textTheme.labelMedium?.copyWith(
                color: textColor,
                fontWeight: FontWeight.bold,
              ),
        ),
      ),
    );
  }
}

class _ExampleCard extends StatelessWidget {
  const _ExampleCard({
    required this.index,
    required this.example,
    required this.isPlaying,
    required this.onPlay,
  });

  final int index;
  final WordExample example;
  final bool isPlaying;
  final VoidCallback onPlay;

  @override
  Widget build(BuildContext context) {
    return PremiumCard(
      padding: const EdgeInsets.all(20),
      child: Row(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          // Index Number
          Padding(
            padding: const EdgeInsets.only(top: 2),
            child: Text(
              '$index.',
              style: Theme.of(context).textTheme.titleMedium?.copyWith(
                    fontWeight: FontWeight.bold,
                    color: Theme.of(context).colorScheme.primary.withOpacity(0.5),
                  ),
            ),
          ),
          const SizedBox(width: 12),
          // Content
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                NemoFuriganaText(
                  text: example.japanese,
                  baseTextStyle: Theme.of(context).textTheme.bodyLarge?.copyWith(
                        fontSize: 17,
                        height: 1.5,
                      ),
                  furiganaTextColor: Theme.of(context).colorScheme.primary.withOpacity(0.7),
                  furiganaTextSize: 10,
                ),
                const SizedBox(height: 8),
                Text(
                  example.chinese,
                  style: Theme.of(context).textTheme.bodyMedium?.copyWith(
                        color: Theme.of(context).colorScheme.onSurfaceVariant,
                        height: 1.5,
                      ),
                ),
              ],
            ),
          ),
          const SizedBox(width: 8),
          // Speaker Button
          NemoSpeakerButton(
            isPlaying: isPlaying,
            size: 44,
            backgroundColor: Theme.of(context).colorScheme.primary.withOpacity(0.05),
            onClick: onPlay,
          ),
        ],
      ),
    );
  }
}

// Removed _ExampleSpeakerButton

File: archive/feature/library/src/main/java/com/jian/nemo/feature/library/presentation/detail/WordDetailScreen.kt
package com.jian.nemo.feature.library.presentation.detail

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.luminance
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.hilt.navigation.compose.hiltViewModel
import com.jian.nemo.core.designsystem.theme.*
import com.jian.nemo.core.domain.model.Word
import com.jian.nemo.core.ui.component.speaker.SpeakerButton

/**
 * 单词详情界面 (UI/UX Pro Max)
 */
@OptIn(androidx.compose.foundation.ExperimentalFoundationApi::class)
@Composable
fun WordDetailScreen(
    onNavigateBack: () -> Unit,
    viewModel: WordDetailViewModel = hiltViewModel()
) {
    val contextIds by viewModel.contextIds.collectAsState()
    val initialWord by viewModel.currentWord.collectAsState()
    val playingAudioId by viewModel.playingAudioId.collectAsState()
    val isDark = MaterialTheme.colorScheme.background.luminance() < 0.5f

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background)
    ) {
        if (contextIds.isEmpty()) {
            // Fallback: Show initial word or loading while fetching context
            if (initialWord != null) {
                WordDetailContent(
                    word = initialWord!!,
                    isDark = isDark,
                    playingAudioId = playingAudioId,
                    onPlayAudio = viewModel::playAudio,
                    onBack = onNavigateBack
                )
            } else {
                Box(
                    modifier = Modifier.fillMaxSize(),
                    contentAlignment = Alignment.Center
                ) {
                    CircularProgressIndicator()
                }
            }
        } else {
            // Swipe Navigation Enabled
            // Calculate initial page based on the starting word ID
            val initialIndex = remember(contextIds, initialWord) {
                val index = contextIds.indexOf(initialWord?.id ?: -1)
                if (index >= 0) index else 0
            }

            val pagerState = androidx.compose.foundation.pager.rememberPagerState(
                initialPage = initialIndex,
                pageCount = { contextIds.size }
            )

            androidx.compose.foundation.pager.HorizontalPager(
                state = pagerState,
                modifier = Modifier.fillMaxSize()
            ) { page ->
                val wordId = contextIds[page]
                // Fetch word for this specific page
                val word by remember(wordId) { viewModel.getWordFlow(wordId) }.collectAsState(initial = null)

                if (word != null) {
                    WordDetailContent(
                        word = word!!,
                        isDark = isDark,
                        playingAudioId = playingAudioId,
                        onPlayAudio = viewModel::playAudio,
                        onBack = onNavigateBack
                    )
                } else {
                     Box(
                        modifier = Modifier.fillMaxSize(),
                        contentAlignment = Alignment.Center
                    ) {
                        CircularProgressIndicator()
                    }
                }
            }
        }
    }
}

@Composable
private fun WordDetailContent(
    word: Word,
    isDark: Boolean,
    playingAudioId: String?,
    onPlayAudio: (String, String?) -> Unit,
    onBack: () -> Unit
) {
    val scrollState = rememberScrollState()

    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(scrollState)
    ) {
    // === 1. Immersive Hero Section ===
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .background(
                    brush = Brush.verticalGradient(
                        colors = if (isDark) listOf(
                            MaterialTheme.colorScheme.primary.copy(alpha = 0.2f),
                            MaterialTheme.colorScheme.background
                        ) else listOf(
                            MaterialTheme.colorScheme.primary.copy(alpha = 0.1f),
                            MaterialTheme.colorScheme.background
                        )
                    )
                )
        ) {
            // Hero Content (Centered)
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .statusBarsPadding() // Add padding for content below status bar
                    .padding(top = 56.dp, bottom = 32.dp, start = 24.dp, end = 24.dp), // Add top padding to clear header
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                // Word (Kanji)
                Text(
                    text = word.japanese,
                    style = MaterialTheme.typography.displayMedium.copy(
                        fontWeight = FontWeight.Bold,
                        letterSpacing = 1.sp
                    ),
                    color = MaterialTheme.colorScheme.onBackground
                )

                Spacer(modifier = Modifier.height(8.dp))

                // Hiragana
                Text(
                    text = word.hiragana,
                    style = MaterialTheme.typography.titleLarge.copy(
                        fontWeight = FontWeight.Medium
                    ),
                    color = MaterialTheme.colorScheme.onBackground.copy(alpha = 0.7f)
                )

                Spacer(modifier = Modifier.height(24.dp))

                // Play Audio Button (Hero Style)
                val wordAudioId = "word_${word.id}"
                SpeakerButton(
                    isPlaying = playingAudioId == wordAudioId,
                    onClick = { onPlayAudio(word.japanese, wordAudioId) },
                    size = 56.dp,
                    backgroundColor = MaterialTheme.colorScheme.primaryContainer,
                    tint = MaterialTheme.colorScheme.onPrimaryContainer
                )
            }

            // Common Header (Overlaid)
            com.jian.nemo.core.ui.component.common.CommonHeader(
                title = "", // Empty title for hero section look
                onBack = onBack,
                backgroundColor = Color.Transparent
            )
        }

        // === 2. Meaning & Tags ===
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 20.dp)
        ) {
            PremiumDetailCard(isDark = isDark) {
                Column(modifier = Modifier.padding(24.dp)) {
                    // Chinese Meaning
                    Text(
                        text = "中文释义",
                        style = MaterialTheme.typography.labelLarge,
                        color = MaterialTheme.colorScheme.primary
                    )
                    Spacer(modifier = Modifier.height(8.dp))
                    Text(
                        text = word.chinese,
                        style = MaterialTheme.typography.headlineSmall.copy(fontWeight = FontWeight.Bold),
                        color = MaterialTheme.colorScheme.onSurface
                    )

                    Spacer(modifier = Modifier.height(20.dp))

                    // Tags Row
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(8.dp)
                    ) {
                        // Level Tag
                        DetailTag(
                            text = word.level,
                            containerColor = NemoPrimary,
                            contentColor = Color.White
                        )

                        // POS Tag
                        word.pos?.let { pos ->
                            DetailTag(
                                text = pos,
                                containerColor = MaterialTheme.colorScheme.secondaryContainer,
                                contentColor = MaterialTheme.colorScheme.onSecondaryContainer
                            )
                        }
                    }
                }
            }

            Spacer(modifier = Modifier.height(24.dp))

            // === 3. Example Sentences ===
            Text(
                text = "例句",
                style = MaterialTheme.typography.titleLarge.copy(fontWeight = FontWeight.Bold),
                color = MaterialTheme.colorScheme.onBackground,
                modifier = Modifier.padding(start = 4.dp, bottom = 12.dp)
            )

            val examples = listOfNotNull(
                if (!word.example1.isNullOrBlank() && !word.gloss1.isNullOrBlank()) Triple(word.example1!!, word.gloss1!!, 1) else null,
                if (!word.example2.isNullOrBlank() && !word.gloss2.isNullOrBlank()) Triple(word.example2!!, word.gloss2!!, 2) else null,
                if (!word.example3.isNullOrBlank() && !word.gloss3.isNullOrBlank()) Triple(word.example3!!, word.gloss3!!, 3) else null
            )

            if (examples.isNotEmpty()) {
                examples.forEach { (sentence, translation, index) ->
                    ExampleCard(
                        sentence = sentence,
                        translation = translation,
                        isDark = isDark,
                        index = index,
                        playingAudioId = playingAudioId,
                        onPlayAudio = onPlayAudio,
                        wordId = word.id
                    )
                    if (index < examples.size) {
                         Spacer(modifier = Modifier.height(12.dp))
                    }
                }
            } else {
                 Text(
                    text = "暂无例句",
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    modifier = Modifier.padding(start = 4.dp)
                )
            }

            Spacer(modifier = Modifier.height(48.dp))
        }
    }
}

/**
 * Premium Card Container
 */
@Composable
private fun PremiumDetailCard(
    isDark: Boolean,
    modifier: Modifier = Modifier,
    content: @Composable ColumnScope.() -> Unit
) {
    val containerColor = if (isDark) MaterialTheme.colorScheme.surfaceContainer else Color.White
    val shadowElevation = if (isDark) 2.dp else 8.dp
    val shadowColor = if (isDark) Color.Black.copy(alpha = 0.3f) else Color.Black.copy(alpha = 0.05f)

    Surface(
        modifier = modifier
            .fillMaxWidth()
            .shadow(
                elevation = shadowElevation,
                shape = RoundedCornerShape(24.dp),
                spotColor = shadowColor,
                ambientColor = shadowColor
            ),
        shape = RoundedCornerShape(24.dp),
        color = containerColor,
        content = { Column(content = content) }
    )
}

/**
 * Detail Tag (Pill Shape)
 */
@Composable
private fun DetailTag(
    text: String,
    containerColor: Color,
    contentColor: Color
) {
    Surface(
        shape = RoundedCornerShape(50),
        color = containerColor,
        modifier = Modifier.height(28.dp)
    ) {
        Box(
            contentAlignment = Alignment.Center,
            modifier = Modifier.padding(horizontal = 12.dp)
        ) {
            Text(
                text = text,
                style = MaterialTheme.typography.labelMedium.copy(fontWeight = FontWeight.Bold),
                color = contentColor
            )
        }
    }
}

/**
 * Example Sentence Card
 */
@Composable
private fun ExampleCard(
    sentence: String,
    translation: String,
    isDark: Boolean,
    index: Int,
    playingAudioId: String?,
    onPlayAudio: (String, String?) -> Unit,
    wordId: Int
) {
    val exampleId = "example_${wordId}_$index"

    PremiumDetailCard(isDark = isDark) {
        Row(
            modifier = Modifier.padding(20.dp),
            verticalAlignment = Alignment.Top
        ) {
            // Index Number
            Text(
                text = "$index.",
                style = MaterialTheme.typography.titleMedium.copy(fontWeight = FontWeight.Bold),
                color = MaterialTheme.colorScheme.primary.copy(alpha = 0.5f),
                modifier = Modifier.padding(top = 2.dp)
            )

            Spacer(modifier = Modifier.width(12.dp))

            Column(modifier = Modifier.weight(1f)) {
                com.jian.nemo.core.ui.component.text.FuriganaText(
                    text = sentence,
                    baseTextStyle = MaterialTheme.typography.bodyLarge.copy(
                        fontSize = 17.sp,
                        fontWeight = FontWeight.Normal, // Changed from Medium to Normal
                        lineHeight = 24.sp
                    ),
                    baseTextColor = MaterialTheme.colorScheme.onSurface,
                    furiganaTextColor = if (isDark) MaterialTheme.colorScheme.primary.copy(alpha = 0.8f) else MaterialTheme.colorScheme.primary.copy(alpha = 0.7f),
                    furiganaTextSize = 10.sp
                )

                Spacer(modifier = Modifier.height(8.dp))

                // Translation
                Text(
                    text = translation,
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }

            Spacer(modifier = Modifier.width(8.dp))

            // Speaker Button
            SpeakerButton(
                isPlaying = playingAudioId == exampleId,
                onClick = { onPlayAudio(sentence, exampleId) },
                size = 44.dp,
                tint = MaterialTheme.colorScheme.primary
            )
        }
    }
}
